tfexec/terraform.go

```diff
@@ -0,0 +1,606 @@
+package tfexec
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+
+	tfjson "github.com/hashicorp/terraform-json"
+)
+
+type Terraform struct {
+	execPath    string
+	workingDir  string
```

Comment:
this should probably be referred to as the `rootModulePath` or something, its the working dir for TF execution, but only because the working dir is assumed to be the root module, so may be a clearer name.

Reply:
I think this will largely depends on the command being executed. Most commands (`apply`, `plan`, `destroy`) do expect the workdir to be also root module, but some do not, so I'd be careful about early (from perspective of the workflow) assumptions here.

Reply:
Agree with Radek here, can disregard my comment.